core/postsolve.gms

```diff
@@ -164,9 +133,13 @@ display s_actualbudgetco2;
 if(cm_iterative_target_adj eq 6,
 *JeS* Update tax levels/ multigasbudget values to reach the peak CO2 budget
  
-p_actualbudgetco2(t) =           sum(ttot$(ttot.val < t.val AND ttot.val > 2010), (sum(regi, (vm_emiTe.l(ttot,regi,"co2") + vm_emiCdr.l(ttot,regi,"co2") + vm_emiMac.l(ttot,regi,"co2"))) * sm_c_2_co2 * pm_ts(ttot)))
-                              + sum(regi, (vm_emiTe.l(t,regi,"co2") + vm_emiCdr.l(t,regi,"co2") + vm_emiMac.l(t,regi,"co2")))*sm_c_2_co2 * (pm_ts(t) * 0.5 + 0.5)
-                              + sum(regi, (vm_emiTe.l("2010",regi,"co2") + vm_emiCdr.l("2010",regi,"co2") + vm_emiMac.l("2010",regi,"co2")))*sm_c_2_co2 * 2;
+*** p_actualbudgetco2 for t > 2020. It includes emissions from 2020 to t (inclusive).
+$offOrder
+p_actualbudgetco2(t)$(t.val > 2020) = sum(ttot$(ttot.val < t.val AND ttot.val > 2020), (sum(regi, (vm_emiTe.l(ttot,regi,"co2") + vm_emiCdr.l(ttot,regi,"co2") + vm_emiMac.l(ttot,regi,"co2"))) * sm_c_2_co2 * pm_ts(ttot)))
+                       + sum(regi, (vm_emiTe.l(t,regi,"co2") + vm_emiCdr.l(t,regi,"co2") + vm_emiMac.l(t,regi,"co2"))) * sm_c_2_co2 * ((pm_ttot_val(t)-pm_ttot_val(t-1))/2 + 0.5)
+                       + sum(regi, (vm_emiTe.l("2020",regi,"co2") + vm_emiCdr.l("2020",regi,"co2") + vm_emiMac.l("2020",regi,"co2"))) * sm_c_2_co2 * (pm_ts("2020") * 0.5 + 0.5);
+$onOrder
```

Comment:
Adding these controls, I felt like hacking, but I couldn't find another way to get t-1 I used in line 139.